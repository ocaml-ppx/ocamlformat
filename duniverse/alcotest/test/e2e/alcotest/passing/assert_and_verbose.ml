(** Test the interaction between ASSERT prints and the `--verbose` option. *)

let () =
  let open Alcotest in
  run ~verbose:true "assert-and-verbose"
    [
      ( "alpha",
        [
          Alcotest.test_case "check → stdout" `Quick (fun () ->
              Alcotest.(check unit) "alpha-0 check" () ();
              Format.printf "alpha-0 standard out\n");
          Alcotest.test_case "stdout → check" `Quick (fun () ->
              Format.printf "\nalpha-1 standard out\n";
              Alcotest.(check unit) "alpha-1 check" () ());
          Alcotest.test_case "check → stderr" `Quick (fun () ->
              Alcotest.(check unit) "alpha-2 check" () ();
              Format.eprintf "alpha-2 standard error\n");
        ] );
      ( "beta",
        [
          Alcotest.test_case "stdout → check → stderr" `Quick (fun () ->
              Format.printf "\nbeta-0 standard out\n";
              Alcotest.(check unit) "beta-0 check" () ();
              Format.eprintf "beta-0 standard error\n");
        ] );
    ]
