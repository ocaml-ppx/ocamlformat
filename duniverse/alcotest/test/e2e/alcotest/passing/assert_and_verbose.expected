Testing `assert-and-verbose'.
This run has ID `<uuid>'.

ASSERT alpha-0 check
alpha-0 standard out
  [OK]          alpha          0   check → stdout.

alpha-1 standard out
ASSERT alpha-1 check
  [OK]          alpha          1   stdout → check.
ASSERT alpha-2 check
alpha-2 standard error
  [OK]          alpha          2   check → stderr.

beta-0 standard out
ASSERT beta-0 check
beta-0 standard error
  [OK]          beta           0   stdout → check → stderr.

Test Successful in <test-duration>s. 4 tests run.
